### ISSUE TITLE
bazel.build website improvements

### ISSUE THREAD

**Issue by laszlocsomor** [bazelbuild/bazel#5208]
I'd like to collect ideas how to improve the bazel.build and docs.bazel.build websites. My main issue is that some things are harder to find or navigate than I think they could be. There are finer, more focused observations too.

I'm opening this bug with the intent of collecting these ideas -- mine and everyone else's -- and use it as a burn-down list of improvements. Please feel free to add your ideas. I'll keep updating the list as tasks are completed.

---

*   Landing page:
    *   "Get Bazel" button:
        *   Rename to "Download Bazel $VERSION for $PLATFORM" to disambiguate meaning and provide direct download link.
        *   Autodetect user's platform, set the button's caption and link accordingly.
        *   Make this button equally prominent as current "Get Started" button.
    *   Rename "Get Started" button to "See Tutorials".
    *   Add a "Take a tour" button left of "Download Bazel" that takes you to a page that introduces Bazel in a nutshell (i.e. one-pager), using a couple screenshots or animations and little text. Reading the entire tour page should require no more than a minute.
    *   Add a non-button link to download Bazel for other platforms, or download older versions.
*   Navigation pane:
    *   Remove the "Experimental Features" page. It contains niche information.
*   Installing on Windows:
    *   Getting Bazel section: make "Download Bazel" a link, pointing to the latest version on GitHub.
*   Build Encyclopedia, User Manual, Query Manual:
    *   Version these with Bazel, allow selecting older versions.

Update(2018/05/16):

- Move the language-specific tutorials to GitHub, link to them from the end of the installation guide.
- Document where to install software, what versions to use.
- Provide project skeletons on GitHub that users can clone and customize. Example of this concept: https://github.com/golang-standards/project-layout
- Document runfiles and the runfile libraries ([#4279](https://github.com/bazelbuild/bazel/issues/4279)).
- Document how to use DLLs on Windows ([#3205](https://github.com/bazelbuild/bazel/issues/3205)).

Update(2018/05/17):

- Move the list of supported rulesets to the top level for improved visibility. Currently, the rulesets are hidden under 2 levels of nesting (Build Encyclopedia > Rules)

Update(2018/06/11):

- from @petemounce: _I would love to see a worked example of how to create a rules library - create a rule, iterate it, test it, ship it, integrate it elsewhere. I'm particularly interested to try to create rules for https://packer.io and https://gohugo.io/ ._

**Comment by laurentlb** [comment]
/cc @jin 

I agree!

**Comment by laszlocsomor** [comment]
Adding to OP:

**Comment by jin** [comment]
Nice! I have another low hanging fruit FR, based off this [Twitter conversation](https://twitter.com/jin_/status/996609549602906113): the list of supported languages are [deeply nested in the sidebar](https://docs.bazel.build/versions/master/bazel-overview.html#be-rules-menu) under "Build Encyclopedia > Rules". I don't think we should hide the polyglot-ness of Bazel, so perhaps promote the list to a top level sidebar section?

Compare this to [Buck's documentation navbar](https://buckbuild.com/setup/getting_started.html), where you can see all of the rulesets at a glance.

**Comment by laszlocsomor** [comment]
@spomorski : Thank you in advance! I believe you have edit rights, feel free to update my OP.

@jin: Good idea! Please update my OP.

**Comment by petemounce** [comment]
I would love to see a worked example of how to create a rules library - create a rule, iterate it, test it, ship it, integrate it elsewhere. I'm particularly interested to try to create rules for https://packer.io and https://gohugo.io/ .

**Comment by HI-MarkPrice** [comment]
One glaring issue on the 'Installing Bazel on Windows' page is that it says you don't need to download a separate JDK 8 install. It also refers to versions on the release page that don't have the JDK integrated -- no such files exist:

```Note: Bazel includes an embedded JDK so you don't need to install one separately. If you want to download Bazel without an embedded JDK, look for bazel-<version>-without-jdk-windows-x86_64.exe on the release page.```

**Comment by laszlocsomor** [comment]
Thanks!

Yes, the page is outdated. There's no JDK-less version anymore. All Windows versions (e.g. https://github.com/bazelbuild/bazel/releases/download/0.15.2/bazel-0.15.2-windows-x86_64.exe) contain an embedded JDK, that's indeed only there to run Bazel itself.

I'll fix these docs tomorrow, or you're welcome to send a PR and assign me as a reviewer if you want to fix it yourself.

**Comment by HI-MarkPrice** [comment]
Another issue I'm having is that the documentation on the website isn't sufficient to get me up and running with Bazel in anything but a trivial project. I'm trying to build a cross platform app and the documentation doesn't tell me clearly how I can make that happen. My latest issue on this is with the apple rules, but in general the examples and tutorials don't really help.

**Comment by HI-MarkPrice** [comment]
BTW, my specific example with the Apple rules is that there is NO example of how to correctly declare the *required* infoplists argument for a macos_application. Really? NO examples? That's just silly.

**Comment by laszlocsomor** [comment]
Pinging Apple team folks: @aragos @c-parsons @rupertks

**Comment by spomorski** [comment]
Should we let Kelvin's team know, too?

**Comment by laszlocsomor** [comment]
Who is Kelvin?

**Comment by spomorski** [comment]
kelvinchan@, his team owns the rules_apple
<https://github.com/bazelbuild/rules_apple> rules.

**Comment by sergiocampama** [comment]
@HI-MarkPrice there are examples in the rules_apple repo: https://github.com/bazelbuild/rules_apple/blob/master/examples/macos/HelloWorld/BUILD 

did you find those are missing anything?

**Comment by HI-MarkPrice** [comment]
Huh. I honestly didn't even see the examples folder inside of the apple_rules repo. I started from the README.md (glossing over the iOS example because I'm not doing iOS) and followed the link to the reference documentation and then onto the Info.Plist documentation at https://github.com/bazelbuild/rules_apple/blob/master/doc/common_info.md#infoplist-handling
I also looked at the actual rule definition for macos_application. Neither of these helped & being a non-Python user I couldn't figure out what to do to make it work, so I reverted back to reading the Skylark language spec & the Concepts and Terminology pages.

It looks like I might be able to get a bit further with the macos example in the repo. Fingers crossed.
